### PR TITLE
[FrameworkBundle][Serializer] Move normalizer/encoders definitions to xml file & remove unnecessary checks

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | 3.3 
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | N/A 
| License       | MIT
| Doc PR        | N/A

- `DataUriNormalizer` was added in 3.1 (https://github.com/symfony/symfony/commit/cc7b5af17019ca941ce1553376aaf6455a1fc919#diff-b7fc65c7d852312152e353f395fc70a8)

- `DateTimeNormalizer` was added in 3.1 (https://github.com/symfony/symfony/commit/6749a70326dcc32ba0d524455f3e245c6a85b385#diff-17828084c07e429d87a1754420d312ef)

- `JsonSerializableNormalizer` was added in 3.1 (https://github.com/symfony/symfony/commit/a6788813fa9973785d81c840ba125b2df0e81571#diff-537ddf87a3d7ff914be4244a1e0c07f0)

- `YamlEncoder` was added in 3.2 (https://github.com/symfony/symfony/commit/9366a7dc774299cdeb722f61349525f5b717e92c#diff-11720cb963c63bb0ad3fb23aba4ae294)
    `Yaml::DUMP_OBJECT` was added in 3.1 (https://github.com/symfony/symfony/commit/286103b2254213eada68517095a6d794b4749d64#diff-5199351a5995f15f224160f6969931c5R23)

- `CsvEncoder` was added in 3.2 (https://github.com/symfony/symfony/commit/e71f5bea961566dd459a65c347e38e1fb800608d#diff-4de6bbbd40ea769ada711de28fb180c8)

while on 3.3, FrameworkBundle [conflicts](https://github.com/symfony/symfony/blob/3.3/src/Symfony/Bundle/FrameworkBundle/composer.json#L70) with `"symfony/serializer": "<3.3"` and both 3.1 and 3.2 are EOL anyway.

Moving these definitions to the `serializer.xml` file unclutters a bit the `FrameworkExtension`, make things clear about these service being always registered and allows the PhpStorm's Symfony plugin to properly detect and jump to them from classes.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
> allows the PhpStorm's Symfony plugin to properly detect and jump to them from classes

which means we should always favor putting services in xml, then removing them when applicable, instead of adding them in the DI ext? We just did that the wrong way recently in another PR, isn't it?

**Comment by ogizanagi** [comment]
> which means we should always favor putting services in xml, then removing them when applicable, instead of adding them in the DI ext?

Yes indeed, would be better.
For instance in https://github.com/symfony/symfony/commit/6185cb19913984c336f7aee389842bed958af73f#diff-2166feba774c51187560f4bac8aa19b3, but it was consistent with the way other normalizers were registered so far.
Did you have another PR in mind?

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @ogizanagi.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml [hunk 1]
```diff
@@ -86,6 +101,14 @@
             <tag name="serializer.encoder" />
         </service>
 
+        <service id="serializer.encoder.yaml" class="Symfony\Component\Serializer\Encoder\YamlEncoder">
```
Comment (GuilhemN) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml]: This is broken, `symfony/yaml` may not be installed and we now have an error when trying to use the serializer... See https://travis-ci.org/nelmio/NelmioApiDocBundle/jobs/305754242.

Comment (ogizanagi) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml]: Hmm. My bad. I'll look at this soon. Thanks for the report!

Comment (GuilhemN) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml]: Thanks!

Comment (ogizanagi) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml]: See https://github.com/symfony/symfony/pull/25179

**Pull request closed without merging**